### ISSUE TITLE
Create patch case to adapt project to a specific case

### ISSUE THREAD

**Issue by lexious89** [platisd/indoor-navigation-system#94]
## Description
As the project is intended for indoor settings, different cases will require special configurations, graphics and resources in local config etc. You can either do this in code or by patching before building.
Consider use case A for a large parking area as compared to a medium 3-story office building usecase. Obviously, these scenarios present different challenges and the INS platform might have to adapt accordingly, if we decide to keep one repo for all these cases. We need to patch the sw before we build. Some 'patchable' options include: the ins node sw (in accordance with the hw- revision), indoor mapping, positioning algorithm and local config params. This however is not the same as the tuning process. Whilst the tuning process is performed after the system is built and flashed on the system, in order to adapt the system to a specific usecase, the patching process is performed before building the sw to align the appropriate different sw resource permutations for a specific application.
We want to create a simple patch for a dummy 'platform' that anyone can modify to suite their own application.
## Definition of Done
Fulfill the above requirement by providing a patch script in the repository.

**Comment by platisd** [comment]
Can you be more specific on what needs to be done? If I were to take up this story I wouldn't know what to do or where to start!

**Comment by platisd** [comment]
I was thinking of this and I believe that you mean that we should refactor this so to be a platform and be able to satisfy different use cases. Right now the "platform" (i.e. purpose agnostic hardware and software) are tightly coupled with the "indoor navigation" use case.

I do agree that this should happen but it's a rather large task that should be broken down further and possibly some architectural changes should ensue!

**Comment by lexious89** [comment]
Yes, I agree the description is quiet vague. I have updated the contents above. The platform is indeed for indoor cases alone. But different indoor settings will employ different wifi APs (mac address, rssi levels etc), and indoor maps, perhaps in the future, various ins node hw/sw-revisions. We can't anticipate all use cases. But it would be really helpful to be able to run a script to swap between various use cases (platform) and a generic patch script might be enough. This is what I meant. 
I guess we could break it down into smaller series of tasks.

**Comment by platisd** [comment]
OK I understand better now but still think it needs smaller tasks as well as some architectural discussion. The main problem is that some of the infrastructure and prerequisites for this are missing.
